src/ops/info.rs

```diff
@@ -29,12 +33,19 @@ pub fn info(spec: &str, config: &Config, reg_or_index: Option<RegistryOrIndex>)
     if let Ok(root) = root_manifest(None, config) {
         let ws = Workspace::new(&root, config)?;
         if let Some(resolve) = ops::load_pkg_lockfile(&ws)? {
-            if let Ok(p) = resolve.query(spec) {
+            if let Ok(p) = resolve.query(spec.name()) {
```

Comment:
You likely want to call `specs_to_ids`.  It can return multiple results, just pick the highest.
- Avoids the need for the `matches` call below
- Picks a value when there is ambiguity, rather than erroring and falling through to the registry (we should have a test case added for this)

Reply:
I am a little bit confused. It seems the `specs_to_ids` API also uses the `query` API internally. 

It can only return one matched package or bail out an error:
```rust
    /// Checks a list of `PackageId`s to find 1 that matches this `PackageIdSpec`. If 0, 2, or
    /// more are found, then this returns an error.
    pub fn query<I>(&self, i: I) -> CargoResult<PackageId>
```

Reply:
Overlooked that.  We likely want to just call `.iter()` and do our own calls to `matches`...